### ISSUE TITLE
Introduce labels

### ISSUE THREAD

**Issue by paolobrasolin** [jonsterling/math#24]
Currently, nodes can be identified by:
* slugs -- stable univoque alphanumeric codes from `0000` to `ZZZZ`
* numberings -- unstable univoque hierarchical numbers just like the ones from technical books (introduced in #23)

These correspond to Stacks's tags and numbers.

The LaTeX document underlying the Stacks website also has labels (as in LaTeX `\label{...}`).
In fact, that's what one uses when _actually editing the document_.

I believe alphanumeric slugs have bad ergonomics and we could make our lives simpler. In short, I'm envisioning a system where I can create a file named `0007-change-of-base.md` and any of the following lines works as expected:
```
@include{0007}
{% ref 0007 %}
@include{change-of-base}
{% ref change-of-base %}
```

On top of that, we could have some comforts which are pretty much expected (*) when working with Jekyll:
* only the alphanumeric code is mandatory in the filename; the label can be generated from the document title
* the title in the front matter is not mandatory and it's generated from the label part of the filename if it's missing

(*): that's possible because of the naming convention `YYYY-MM-DD-kebab-case-title.md` for posts

I also suggest taking advantage of the young age of the project and rename our "slug" entity, since "slug" is an already common notion in Jekyll which corresponds to the "label" I'm suggesting.

So, to sum it up with an example:

|                          | Sheafy    | Gerby  |
| ------------------------ | --------- | ------ |
| `0007-change-of-base.md` | filename  | --     |
| `0007`                   | uid       | tag    |
| `change-of-base`         | slug      | label  |
| 2.9                      | numbering | number |

where uid and slug can be used interchangeably when including or referencing a node.

I think would be a nice QOL improvement. What do you think @jonsterling?

**Comment by jonsterling** [comment]
So personally I actually prefer the pure alphanumerics for my own use-case, but I understand that others may not agree. Therefore I am happy with your proposal so long as it remains possible for me to ignore the labels. My one suggestion is that if you do add labels, that you do not generate them from the document title and instead require them to be explicitly named when used. This is because section titles change quite often in the course of ordinary writing, but labels are at least in principle more likely to be stable.

By the way, my reason for preferring the uids, even though it feels counterintuitive, is that these are truly stable --- even if it less common, one can easily imagine many "refactorings" that change the label.

**Comment by jonsterling** [comment]
(Btw, I'm also happy with your updated `slug` vs `uid` ontology)

**Comment by jonsterling** [comment]
PS i also have no problem calling our uids tags, the only reason i didn’t was to avoid confusion with tags in the sense of blogging. anyway i just wanted to bring that up in case you prefer it. personally i don’t have a preference

**Comment by paolobrasolin** [comment]
> PS i also have no problem calling our uids tags, the only reason i didn’t was to avoid confusion with tags in the sense of blogging. anyway i just wanted to bring that up in case you prefer it. personally i don’t have a preference

Ah, yes, that's the exact reason why I avoided calling them "tags".

**Comment by paolobrasolin** [comment]
I'll keep track of the idea in paolobrasolin/jekyll-sheafy#7, so I'm closing this for the sake of decluttering.
